File: Algorithm/Python/492. Construct the Rectangle.py
# -*- coding: utf-8 -*-
# @Author: WuLC
# @Date:   2017-01-24 00:02:10
# @Last modified by:   WuLC
# @Last Modified time: 2017-01-24 00:02:23
# @Email: liangchaowu5@gmail.com


class Solution(object):
    def constructRectangle(self, area):
        """
        :type area: int
        :rtype: List[int]
        """
        for i in reversed(xrange(1, int(math.sqrt(area))+1)):
            if area%i == 0:
                return [area/i, i]
        

File: Algorithm/Java/492. Construct the Rectangle.java
/**
* Author: WuLC
* Date:   2017-05-04 15:07:29
* Last modified by:   WuLC
* Last Modified time: 2017-05-04 15:08:47
* Email: liangchaowu5@gmail.com
*/


// start testing from sqrt root of the area
public class Solution 
{
    public int[] constructRectangle(int area) 
    {
        for(int i = (int)Math.sqrt(area); i <= area; i++)
        {
            if( area % i == 0)
            {
                int j = area/i;
                return new int[]{Math.max(i, j), Math.min(i, j)};
            }
        }
        return new int[2];
    }
}